dataflow/src/main/java/org/checkerframework/dataflow/cfg/AbstractCFGVisualizer.java

```diff
@@ -204,20 +201,17 @@ protected String visualizeBlockHelper(
                     @SuppressWarnings("nullness:contracts.precondition.not.satisfied")
                     S store = analysis.getResult().getStoreAfter(lastNode);
                     StringBuilder sbStore = new StringBuilder();
-                    sbStore.append(escapeString).append("~~~~~~~~~").append(escapeString);
+                    sbStore.append("~~~~~~~~~").append(escapeString);
                     sbStore.append("After: ");
                     if (store != null) {
                         sbStore.append(visualizeStore(store));
                     } else {
-                        sbStore.append("null store");
+                        sbStore.append("null store").append(escapeString);
                     }
                     sbBlock.append(sbStore);
                 }
             }
         }
-        if (!centered) {
```

Comment:
What happens if analysis is `null`? Before, it looks like an `escapeString` is appended when it is a `SPECIAL_BLOCK`.
Is that missing now? Is that a problem or intended?

Reply:
It is intended, this logic is no longer needed.

`visualizeSpecialBlockHelper` should be responsible for making special blocks centered. `DOTCFGVisualizer` now passes a separator to `visualizeSpecialBlockHelper`: https://github.com/typetools/checker-framework/pull/3395/files#diff-f708a265ff3b24e68538c35aef4dfb7eR136

Now in this situation, when the analysis is null, a special block in DOT will be like: `[shape=oval label="<entry>\n"]` (already centered).